locations/spiders/tesla.py

```diff
@@ -1,88 +1,154 @@
-
-        if "supercharger" in location_data.get("location_type"):
-            apply_category(Categories.CHARGING_STATION, feature)
-            feature["brand_wikidata"] = "Q17089620"
-            feature["brand"] = "Tesla Supercharger"
-
-            # Capture capacity of the supercharger
-            regex = r"(\d+) Superchargers, available 24\/7, up to (\d+kW)(<br />CCS Compatibility)?"
-            regex_matches = re.findall(regex, location_data.get("chargers"))
-            if regex_matches:
-                for match in regex_matches:
-                    capacity, output, ccs_compat = match
-
-                    if ccs_compat:
-                        feature["extras"]["socket:type2_combo"] = capacity
-                        feature["extras"]["socket:type2_combo:output"] = output
-                    else:
-                        feature["extras"]["socket:nacs"] = capacity
-                        feature["extras"]["socket:nacs:output"] = output
-
-        if "tesla_center_delivery" in location_data.get("location_type"):
-            apply_category(Categories.SHOP_CAR, feature)
-
-        if "store" in location_data.get("location_type"):
-            apply_category(Categories.SHOP_CAR, feature)
-
-        if "service" in location_data.get("location_type"):
-            apply_category(Categories.SHOP_CAR_REPAIR, feature)
-
-        yield feature
+        return feature
+
+    def parse_phone(self, location_type: str, location: dict) -> str | None:
+        phones = location.get("sales_phone")
+        if not isinstance(phones, list):
+            return None
+        return next((phone.get("number") for phone in phones if phone.get("type") == location_type), None)
+
+    def parse_url(self, location_type: str, location: dict) -> str:
+        return f"https://www.tesla.com/findus/location/{location_type}/{location.get('location_id')}".replace(" ", "")
+
+    def parse_hours(self, hours: str) -> OpeningHours:
+        oh = OpeningHours()
+        clean_html = unescape(hours)
+        rows = Selector(text=clean_html).xpath("//table/tr")
+        for row in rows:
+            day = row.xpath("td[1]/text()").get()
+            hours = row.xpath("td[2]/text()").get()
+            oh.add_ranges_from_string(f"{day}: {hours}")
+        return oh
+
+    # Skip destination chargers as they're not Tesla-operated
+    # Skip if "Coming Soon" - no content to capture yet
+    # Selection only those in categories list
+    def select_locations(self, locations: List[dict]) -> List[dict]:
+        categories = ["store", "service", "bodyshop", "supercharger"]
+        return list(
+            filter(
+                lambda location: location.get("open_soon") != "1"
+                and "destination charger" not in location.get("location_type", [])
+                and any(category in location.get("location_type", []) for category in categories),
+                locations,
+            )
+        )
```

Comment:
Will be a bit cleaner if list comprehension is used instead of filter + lambda:
```suggestion
        return [
            location for location in locations
            if (location.get("open_soon") != "1" and
            "destination charger" not in location.get("location_type", []) and
            any(category in location.get("location_type", []) for category in categories))
         ]
```